utils/flash_access.c

```diff
@@ -0,0 +1,124 @@
+#define CONFIG_SPI_FLASH_NO_FAST_READ
+
+#include <spi/spi_flash.h>
+#include <flash_access.h>
+
+#define FLASH_SIZE_CHUNK   0x1000 //4k
+
+static struct spi_flash *flash_device;
+
+/*******************************************************************************/
+inline int init_flash(void)
+{
+	flash_device = spi_flash_probe(0, 0, 0, 0);
+
+	if (!flash_device)
+		return -1;
+
+	return 0;
+}
+
+/*******************************************************************************/
+inline int is_flash_locked(void)
+{
+	return spi_flash_is_locked(flash_device);
+}
+
+/*******************************************************************************/
+inline int lock_flash(void)
+{
+	return spi_flash_lock(flash_device);
+}
+
+/*******************************************************************************/
+inline int unlock_flash(void)
+{
+	return spi_flash_unlock(flash_device);
+}
+
+inline int read_sec_status(void)
+{
+	return spi_flash_sec_sts(flash_device);
+}
+
+inline int read_sec(u8 reg, u8 addr, void *buf, size_t len)
+{
+	u32 offset = reg;
+	offset <<= 12;
+	offset |= addr;
+
+	return spi_flash_sec_read(flash_device, offset, len, buf);
+}
+
+inline int prog_sec(u8 reg, u8 addr, const void *buf, size_t len)
+{
+	u32 offset = reg;
+	offset <<= 12;
+	offset |= addr;
+
+	return spi_flash_sec_prog(flash_device, offset, len, buf);
+}
+
+inline int lock_sec(u8 reg)
+{
+	return spi_flash_sec_lock(flash_device, reg);
+}
+
+/*******************************************************************************/
+void save_flash(int flash_address, char buffer[MAX_DEVICES][MAX_LENGTH], u8 max_lines, u8 spi_wp_toggle) {
+	int i = 0;
+	int k = 0;
+	int j, ret;
+	char cbfs_formatted_list[MAX_DEVICES * MAX_LENGTH];
+	u32 nvram_pos;
+
+	// compact the table into the expected packed list
+	for (j = 0; j < max_lines; j++) {
+		k = 0;
+		while (1) {
+			cbfs_formatted_list[i++] = buffer[j][k];
+			if (buffer[j][k] == NEWLINE )
```

Comment:
@mek-x looks like asking for troubles. Why we not tracking `k` for some max value ?

Reply:
yeah, I'm changing it to for loop